ostree/ostree_dest.go

```diff
@@ -0,0 +1,245 @@
+package ostree
+
+import (
+	"bytes"
+	"io"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"strings"
+	"strconv"
+	"fmt"
+	"path/filepath"
+
+	"encoding/json"
+	"github.com/containers/image/manifest"
+	"github.com/containers/storage/pkg/archive"
+	"github.com/containers/image/types"
+	"github.com/opencontainers/go-digest"
+	"github.com/pkg/errors"
+)
+
+type blobToImport struct {
+	Size      int64
+	Digest    digest.Digest
+	BlobPath  string
+	Hash      string
+}
+
+type descriptor struct {
+	Size      int64         `json:"size"`
+	Digest    digest.Digest `json:"digest"`
+}
+
+type manifestSchema struct {
+	ConfigDescriptor  descriptor        `json:"config"`
+	LayersDescriptors []descriptor      `json:"layers"`
+}
+
+type ostreeImageDestination struct {
+	ref ostreeReference
+	manifest string
+	schema manifestSchema
+	blobs map[string]*blobToImport
+}
+
+// newImageDestination returns an ImageDestination for writing to an existing ostree.
+func newImageDestination(ref ostreeReference) types.ImageDestination {
+	return &ostreeImageDestination{ref, "", manifestSchema{}, map[string]*blobToImport{}}
+}
+
+// Reference returns the reference used to set up this destination.  Note that this should directly correspond to user's intent,
+// e.g. it should use the public hostname instead of the result of resolving CNAMEs or following redirects.
+func (d *ostreeImageDestination) Reference() types.ImageReference {
+	return d.ref
+}
+
+// Close removes resources associated with an initialized ImageDestination, if any.
+func (d *ostreeImageDestination) Close() {
+}
+
+func (d *ostreeImageDestination) SupportedManifestMIMETypes() []string {
+	return []string{
+		manifest.DockerV2Schema2MediaType,
+	}
+}
+
+// SupportsSignatures returns an error (to be displayed to the user) if the destination certainly can't store signatures.
+// Note: It is still possible for PutSignatures to fail if SupportsSignatures returns nil.
+func (d *ostreeImageDestination) SupportsSignatures() error {
+	return nil
+}
+
+// ShouldCompressLayers returns true iff it is desirable to compress layer blobs written to this destination.
+func (d *ostreeImageDestination) ShouldCompressLayers() bool {
+	return false
+}
+
+// AcceptsForeignLayerURLs returns false iff foreign layers in manifest should be actually
+// uploaded to the image destination, true otherwise.
+func (d *ostreeImageDestination) AcceptsForeignLayerURLs() bool {
+	return false
+}
+
+func (d *ostreeImageDestination) PutBlob(stream io.Reader, inputInfo types.BlobInfo) (types.BlobInfo, error) {
+	hash := strings.Replace(inputInfo.Digest.Hex(), "sha256:", "", 1)
```

Comment:
> Isn’t there a native method to get the hex value without the algorithm ID?

That’ method is `.Hex()` already, actually. So, that `strings.Replace()` should never be needed AFAICT; if it is, something is going wrong elsewhere.